core-tests/src/test/java/org/mule/runtime/core/DefaultEventContextTestCase.java

```diff
@@ -421,6 +421,57 @@ public void externalCompletionWithChild() throws Exception {
     assertCompletionDone(parent);
   }
 
+  @Test
+  @Description("When a child event context is serialized the parent context no longer waits for completion of child context.")
```

Comment:
This is wrong. Just because the event is serialized in  MP-N, it doesn't mean that MP-N+1 will continue processing it and the parent does need to wait for its completion.